src/cache/local_cache.cpp

```diff
@@ -368,6 +368,13 @@ void local_cache_t::get_file(const hasher_t::hash_t& hash,
   } else {
     file::copy(source_path, target_path);
   }
+
+#ifdef _WIN32
+  // Open the target file for writing without doing any actual writes.
+  // This forces the registration of `target_path` as written by MSBuild
+  // FileTracker - i.e. adds an entry in *.write.*.tlog files.
+  file::append("", target_path);
```

Comment:
I'm a bit uncomfortable with this solution, though I can't really think of a better one a.t.m.

* `comp::decompress_file()` and `file::copy()` use the tmp+move trick, but `file::link_or_copy()` doesn't. The latter does a touch. Would that be enough (instead of append)?
* I'm generally allergic to treating _WIN32 == we're using MSVC, but passing around extra wrapper-specific state (some kind of `needs_msvc_filetracker_hack`) seems to be overkill. Funny thing: The MSVC wrapper is built and used on Linux and MacOS too - though I have not tried to make use of it (not sure if it's possible, e.g. via wine).

Reply:
I agree that this is a somewhat hacky solution. We had tried the touch you mentioned, i.e. calling `_wutime64()`, alas it's not caught by FileTracker. Now thinking about it, buildcache process could detect if it's under tracking - two solutions:
* Check if the name of the parent process is `Tracker.exe`, or
* Check if any of the following is loaded: `FileTracker.dll`, `FileTracker32.dll`, `FileTracker64.dll`. 

The latter seems to be easier to implement, with a simple call to [`GetModuleHandleW()`](https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulehandlew).

Reply:
I think that I like that idea. It would definitely motivate an `#ifdef _WIN32`.

While we're at it... I had a quick look at the FileTracker API (https://docs.microsoft.com/en-us/visualstudio/msbuild/file-tracking?view=vs-2019) - though I have to admit that I don't really understanding the scope of the API (e.g. what contexts are live when and where, and if it's possible to affect a "parent" context initiated by MSBuild for instance).

Question: Would there be a benefit to turning off file tracking during certain BuildCache tasks (e.g. preprocessing and cache lookup) - assuming that it's possible? E.g. for performance reasons, or to avoid confusing the build system.

Reply:
A correction here: we did try touch by using Win32 API `SetFileTime()`, and not by calling `_wutime64()`. While the former "escapes" FileTracker, it turns out the latter doesn't. Therefore I think the cleanest solution is to replace this dummy `append()` behind `_WIN32` with an unconditional `touch()`.

Reply:
On the FileTracker API: we experimented with that too - finding the .dll and suspending the tracking - but in the end we found that moving the BuildCache directory to an ignored location is much simpler / less error-prone. And it does exactly what we need: ignore reads and writes from and to the cache.